gpiozero/output_devices.py

```diff
@@ -562,8 +562,12 @@ def __init__(
             raise GPIOPinMissing('red, green, and blue pins must be provided')
         LEDClass = PWMLED if pwm else LED
         super(RGBLED, self).__init__()
-        self._leds = tuple(LEDClass(pin, active_high) for pin in (red, green, blue))
-        self.value = initial_value
+        try:
+            self._leds = tuple(LEDClass(pin, active_high) for pin in (red, green, blue))
+            self.value = initial_value
+        except:
+            self.close()
+            raise
```

Comment:
Hmm, actually looking at it ... it's a little more complex than I thought because some Device references will be lost already in various comprehensions. Hmm. Might be time to tackle the "internal" references that keep certain things like Button alive without an external reference. I'll dig deeper into this tonight.